File: threads.go
package main

import (
	"fmt"
	"time"
)

type Rebel struct {
	data      string
	recipient int
}

func initial(ch chan Rebel, rebel Rebel) {
	fmt.Println("initial thread")
	ch <- rebel
}

func thread(ch chan Rebel, num int) {
	fmt.Println("thread", num) 
	token := <-ch
	if token.recipient == num {
		fmt.Println(rebel.data, "received")
	} else {
		ch <- token
	}
}

func main() {
	n := 6
	rebel := Rebel{"data", n}
	ch := make(chan Rebel)
	go initial(ch, rebel)
	for i := 1; i <= n; i++ {
		go thread(ch, i)
	}
	time.Sleep(time.Second)
}

File: threads.cs
using System;
using System.Text;
using System.Threading;


class Program
{
  const int n = 20;
  static int c = 0;
  static void Threading(int i, Token t)
  {
    if (i<n)
    {
      Console.WriteLine(i+ " thread initialized");
      Thread thread = new Thread(() => Threading(i + 1, t));
      thread.Start();
      Thread.Sleep(n);
      if (t.Recipient == i)
      {
        Console.WriteLine(t.Data + " received in " + i + " thread");
        c=1;
      }
    }
  }
  static void Main(string[] args)
  {
    Console.WriteLine("vvedite celevoi potok");
    int k=int.Parse(Console.ReadLine());
    Token token=new Token(){Recipient = k, Data = "data"};
    Thread thread = new Thread(() => { Threading(1, token); });
    thread.Start();
    Thread.Sleep(n*2+1);
    if (c==0){Console.WriteLine("Povorot ne tuda");}
  }
}
public class Token
{
  public string Data;
  public int Recipient; 
 
}
